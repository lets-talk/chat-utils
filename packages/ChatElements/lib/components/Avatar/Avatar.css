@import './config.css';

.letstalk-avatar-container {
    position: relative;
    overflow: hidden;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow: hidden;
}

.letstalk-avatar-container .letstalk-avatar-image-container {
    position: relative;
}

.letstalk-avatar-container .letstalk-avatar-image-container .letstalk-avatar-image {
    width: 100%;
    height: 100%;
    background-color: rgb(188, 188, 188);
}

.letstalk-avatar-container.rounded .letstalk-avatar-image {
    border-radius: 5px;
}

.letstalk-avatar-container.circle .letstalk-avatar-image {
    border-radius: 100%;
}

.letstalk-avatar-container .letstalk-avatar-image-div {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.letstalk-avatar-container.flexible .letstalk-avatar-image {
    height: auto !important;
    width: 100% !important;
    border-radius: unset !important;
    overflow: unset !important;
}

.letstalk-avatar-container.xsmall {
    width: var(--avatar-size-xsmall);
    height: var(--avatar-size-xsmall);
}

.letstalk-avatar-container.xsmall .letstalk-avatar-image{
    width: var(--avatar-size-xsmall);
    height: var(--avatar-size-xsmall);
}

.letstalk-avatar-container.xsmall .letstalk-avatar-status {
  width: calc(var(--avatar-size-xsmall) / 10);
  height: calc(var(--avatar-size-xsmall) / 10);
  bottom: calc(var(--avatar-size-xsmall) / 13);
  right: calc(var(--avatar-size-xsmall) / 5);
}

.letstalk-avatar-container.small {
    width: var(--avatar-size-small);
    height: var(--avatar-size-small);
}

.letstalk-avatar-container.small .letstalk-avatar-image{
    width: var(--avatar-size-small);
    height: var(--avatar-size-small);
}

.letstalk-avatar-container.small .letstalk-avatar-status {
  width: calc(var(--avatar-size-small) / 10);
  height: calc(var(--avatar-size-small) / 10);
  bottom: calc(var(--avatar-size-small) / 13);
  right: calc(var(--avatar-size-small) / 5);
}

.letstalk-avatar-container.medium {
    width: var(--avatar-size-medium);
    height: var(--avatar-size-medium);
}

.letstalk-avatar-container.medium .letstalk-avatar-image{
    width: var(--avatar-size-medium);
    height: var(--avatar-size-medium);
}

.letstalk-avatar-container.medium .letstalk-avatar-status {
  width: calc(var(--avatar-size-medium) / 10);
  height: calc(var(--avatar-size-medium) / 10);
  bottom: calc(var(--avatar-size-medium) / 13);
  right: calc(var(--avatar-size-medium) / 5);
}

.letstalk-avatar-container.large {
    width: var(--avatar-size-large);
    height: var(--avatar-size-large);
}

.letstalk-avatar-container.large .letstalk-avatar-image{
    width: var(--avatar-size-large);
    height: var(--avatar-size-large);
}

.letstalk-avatar-container.large .letstalk-avatar-status {
  width: calc(var(--avatar-size-large) / 10);
  height: calc(var(--avatar-size-large) / 10);
  bottom: calc(var(--avatar-size-large) / 13);
  right: calc(var(--avatar-size-large) / 5);
}

.letstalk-avatar-container.xlarge {
    width: var(--avatar-size-xlarge);
    height: var(--avatar-size-xlarge);
}

.letstalk-avatar-container.xlarge .letstalk-avatar-image{
    width: var(--avatar-size-xlarge);
    height:var(--avatar-size-xlarge);
}

.letstalk-avatar-container.xlarge .letstalk-avatar-status {
  width: calc(var(--avatar-size-xlarge) / 10);
  height: calc(var(--avatar-size-xlarge) / 10);
  bottom: calc(var(--avatar-size-xlarge) / 13);
  right: calc(var(--avatar-size-xlarge) / 5);
}

.letstalk-avatar-status {
  position: absolute;
  right: 0px;
  border-radius: 100%;
  padding: 1px;
  font-weight: 600;
  border: 2px solid #FFFFFF;
}

.letstalk-avatar-status.online {
  background-color: var(--avatar-status-online);
}

.letstalk-avatar-status.offline {
  background-color: var(--avatar-status-offline);
}

.letstalk-avatar-status.sleeping {
  background-color: var(--avatar-status-sleeping);
}
