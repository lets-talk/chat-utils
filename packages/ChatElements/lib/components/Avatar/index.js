import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './index.scss';

const Avatar = (props) =>
  (
    <div className={classnames('letstalk-avatar-container', props.type, props.size, props.className)}>
      <div className="letstalk-avatar-inner">
        <div className="letstalk-avatar-image-container">
          <img alt={props.alt} src={props.src} className="letstalk-avatar-image" />
          <div className="letstalk-avatar-image-div"></div>
          {props.sideElement}
        </div>
      </div>

      {props.withStatus && <div className={classnames('letstalk-avatar-status', props.status)}>&nbsp;</div>}
    </div>
  );

Avatar.defaultProps = {
  type: 'circle',
  size: 'medium',
  withStatus: false,
  status: '',
  src: '',
  alt: '',
};

Avatar.propTypes = {
  /**
   * types: default, circle, rounded(border radius 5px), flexible
   */
  type: PropTypes.string,
  /**
   * image size. default (25px), xsmall(30px), small(35px), medium(40px), large(45px), xlarge (55px)
   */
  size: PropTypes.string,
  /**
   * image src attribute.
   */
  src: PropTypes.string,
  /**
   * Status choose wheter or not show an status indicator.
   */
  withStatus: PropTypes.bool,
  /**
   * Status choose what kind of status to show. Currently supports: ['online', 'offline', 'sleeping'].
   */
  status: PropTypes.string,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
  /**
   * Image alt attribute.
   */
  alt: PropTypes.object,
};

export default Avatar;
