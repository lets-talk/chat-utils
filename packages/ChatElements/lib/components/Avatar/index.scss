@import './config.css';

// This is used to generate differnet avatar sizes
@mixin avatarStyle($size) {
  .letstalk-avatar-container.#{$size} {
      width: var(--avatar-size-#{$size});
      height: var(--avatar-size-#{$size});
  }

  .letstalk-avatar-container.#{$size} .letstalk-avatar-image{
      width: calc(var(--avatar-size-#{$size}) - 5px);
      height: calc(var(--avatar-size-#{$size}) - 5px);
  }

  .letstalk-avatar-container.#{$size} .letstalk-avatar-status {
    width: calc(var(--avatar-size-#{$size}) / 10);
    height: calc(var(--avatar-size-#{$size}) / 10);
    bottom: calc(var(--avatar-size-#{$size}) / 13);
    right: calc(var(--avatar-size-#{$size}) / 9);
  }
}

.letstalk-avatar-container {
  position: relative;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
  overflow: hidden;

  .letstalk-avatar-image-container {
    position: relative;

    .letstalk-avatar-image-div {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }

    .letstalk-avatar-image {
      width: 100%;
      height: 100%;
      background-color: rgb(188, 188, 188);
    }
  }
}

.letstalk-avatar-container.rounded {
  .letstalk-avatar-image {
    border-radius: 5px;
  }
}

.letstalk-avatar-container.circle {
  .letstalk-avatar-image {
      border-radius: 50%;
  }
}

@include avatarStyle('xsmall');
@include avatarStyle('small');
@include avatarStyle('medium');
@include avatarStyle('large');
@include avatarStyle('xlarge');

.letstalk-avatar-status {
  position: absolute;
  right: 0px;
  border-radius: 100%;
  padding: 1px;
  font-weight: 600;
  border: 2px solid #FFFFFF;
}

.letstalk-avatar-status.online {
  background-color: var(--avatar-status-online);
}

.letstalk-avatar-status.offline {
  background-color: var(--avatar-status-offline);
}

.letstalk-avatar-status.sleeping {
  background-color: var(--avatar-status-sleeping);
}
