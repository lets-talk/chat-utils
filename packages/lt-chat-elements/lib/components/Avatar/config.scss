@import "../../common/styles/variables";

$avatar-size-xsmall: calc(4 * #{$unit});
$avatar-size-small: calc(5 * #{$unit});
$avatar-size-medium: calc(7 * #{$unit});
$avatar-size-large: calc(13 * #{$unit});
$avatar-size-xlarge: calc(21 * #{$unit});

$avatar-status-online: $palette-green-600;
$avatar-status-offline: $palette-red-600;
$avatar-status-sleeping: $palette-yellow-600;
