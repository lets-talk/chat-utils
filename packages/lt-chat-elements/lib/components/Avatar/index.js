import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'styled-components';
import { avatarStyle } from '../../utils/style';

const singleBallBeat = keyframes`{
  0% {
    transform: scale(1);
    opacity: 0.8;
  }

  50% {
    transform: scale(1.5);
    opacity: 1;
  }

  100% {
    transform: scale(1);
    opacity: 0.8;
  }
}`;

const AvatarContainer = styled.div`
  ${(props) => avatarStyle('container', props.size)}
  position: relative;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
  overflow: hidden;
`;

const ImageContainer = styled.div`
  position: relative;
`;

const Image = styled.img`
  width: 100%;
  height: 100%;
  background-color: rgb(188, 188, 188);
  ${(props) => avatarStyle('image', props.size)}
  border-radius: ${(props) =>
    (props.type === 'rounded' && '5px')
    || (props.type === 'circle' && '50%')
    || '0'
};`;

const ImageInnerDiv = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const AvatarStatus = styled.div`
  position: absolute;
  right: 0px;
  border-radius: 100%;
  padding: 1px;
  font-weight: 600;
  border: 2px solid #FFFFFF;
  ${(props) => avatarStyle('status', props.size)}
  ${(props) =>
    props.status && props.status === 'live' &&
      css`
        animation: ${singleBallBeat} 3s infinite;
      `};
  background-color: ${(props) => {
    if (props.status === 'live') return props.theme.palette.common.online;
    if (props.status === 'online') return props.theme.palette.common.online;
    if (props.status === 'offline') return props.theme.palette.common.offline;
    if (props.status === 'sleeping') return props.theme.palette.common.sleeping;
    return props.theme.palette.common.sleeping;
  }}
`;


const Avatar = (props) =>
  (
    <AvatarContainer size={props.size} className={props.className}>
      <ImageContainer>
        <Image alt={props.alt} src={props.src} size={props.size} type={props.type} />
        <ImageInnerDiv ></ImageInnerDiv>
      </ImageContainer>

      {props.withStatus && <AvatarStatus size={props.size} status={props.status}>&nbsp;</AvatarStatus>}
    </AvatarContainer>
  );

Avatar.defaultProps = {
  type: 'circle',
  size: 'medium',
  withStatus: false,
  status: '',
  className: 'LT-Avatar-Container',
  src: '',
  alt: '',
};

Avatar.propTypes = {
  /**
   * types: default, circle, rounded(border radius 5px), flexible
   */
  type: PropTypes.string,
  /**
   * image size. default (25px), xsmall(30px), small(35px), medium(40px), large(45px), xlarge (55px)
   */
  size: PropTypes.string,
  /**
   * image src attribute.
   */
  src: PropTypes.string,
  /**
   * Status choose wheter or not show an status indicator.
   */
  withStatus: PropTypes.bool,
  /**
   * Status choose what kind of status to show. Currently supports: ['online', 'offline', 'sleeping'].
   */
  status: PropTypes.string,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
  /**
   * Image alt attribute.
   */
  alt: PropTypes.string,
};

export default Avatar;
