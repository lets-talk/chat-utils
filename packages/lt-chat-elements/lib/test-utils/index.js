// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
