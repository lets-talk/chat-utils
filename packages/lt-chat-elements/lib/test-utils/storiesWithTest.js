import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
