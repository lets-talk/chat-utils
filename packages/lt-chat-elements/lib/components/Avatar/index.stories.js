import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select, text } from '@storybook/addon-knobs';

// Component to show on storybook
import { Avatar } from '../../../lib';
import { WrapWithTheme } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('Avatar', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('Avatar'));

stories.addWithInfo(
  'Avatar',
  'This is the basic usage with the Avatar with providing a label to show the text.',
  () => (
    <WrapWithTheme
      themeName={select('Theme', { default: 'DefaultTheme', light: 'LightTheme', dark: 'DarkTheme' }, 'default')}
    >
      <Avatar
        src="http://i46.tinypic.com/sexbb8.png"
        className={text('Classname')}
        type={select('Type', { default: 'default', rounded: 'rounded', circle: 'circle' }, 'circle')}
        size={select('Size', {
          xsmall: 'xsmall', small: 'small', medium: 'medium', large: 'large', xlarge: 'xlarge',
        }, 'medium')}
        withStatus={boolean('Show Status', false)}
        status={select('Status', {
          online: 'online', offline: 'offline', sleeping: 'sleeping', live: 'live',
        }, 'online')}
      />
    </WrapWithTheme>
  )
);
