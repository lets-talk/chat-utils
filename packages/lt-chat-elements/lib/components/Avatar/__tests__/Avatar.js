import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { shallowWithTheme, mountWithTheme } from '../../../test-utils';
import Avatar from '../index';

describe('Avatar component', () => {
  it('should render without data', () => {
    const component = shallow(<Avatar />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data and circle case', () => {
    const props = {
      type: 'circle',
      size: 'small',
      withStatus: true,
      status: 'online',
      src: 'http://i46.tinypic.com/sexbb8.png',
      alt: 'An avatar image',
    };

    const component = shallow(<Avatar {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data and children case (String children)', () => {
    const props = {
      type: 'circle',
      size: 'xsmall',
      withStatus: true,
      status: 'live',
      src: 'http://i46.tinypic.com/sexbb8.png',
      alt: 'An avatar image',
      color: '#FEFEFE',
    };

    const component = shallowWithTheme(<Avatar {...props}>S</Avatar>);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.length).toBe(1);
    expect(component.find('StyledChildrenContainer').childAt(0).text()).toBe('S');
    expect(component.find('StyledAvatarStatus').props().size).toBe('xsmall');
    expect(component.find('StyledAvatarStatus').props().status).toBe('live');
  });

  it('should render with data and children case (Component children)', () => {
    const props = {
      type: 'circle',
      size: 'medium',
      withStatus: true,
      status: 'online',
      src: 'http://i46.tinypic.com/sexbb8.png',
      alt: 'An avatar image',
      color: '#FEFEFE',
    };

    const SimpleComponent = () => <h1>Simple component</h1>;

    const component = shallowWithTheme(<Avatar {...props}><SimpleComponent /></Avatar>);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.length).toBe(1);
    // Expect the component to render the <SimpleComponent>
    expect(component.find('StyledChildrenContainer').find('SimpleComponent').length).toBe(1);
    // Expect the component have the proper props()
    expect(component.find('StyledAvatarStatus').props().size).toBe('medium');
    expect(component.find('StyledAvatarStatus').props().status).toBe('online');
  });

  it('should mount with data and children case (Component children)', () => {
    let newProps;
    let component;
    const props = {
      type: 'default', // Test using default case
      size: 'xlarge',
      withStatus: false,
      status: 'online',
      src: 'http://i46.tinypic.com/sexbb8.png',
      alt: 'An avatar image',
      color: '#FEFEFE',
    };

    const SimpleComponent = () => <h1>Simple component</h1>;

    component = mountWithTheme(<Avatar {...props}><SimpleComponent /></Avatar>);
    expect(component.length).toBe(1);
    // Expect the component to render the <SimpleComponent> and have the content we passed
    expect(component.find('StyledChildrenContainer').find('SimpleComponent').length).toBe(1);
    expect(component.find('StyledChildrenContainer').text()).toContain('Simple component');

    // Now the same but rounded
    newProps = { ...props, type: 'rounded' };
    component = mountWithTheme(<Avatar {...newProps}><SimpleComponent /></Avatar>);
    expect(component.length).toBe(1);
    // TODO -> Assert something specific here

    // Now the same but circle
    newProps = { ...props, type: 'circle' };
    component = mountWithTheme(<Avatar {...newProps}><SimpleComponent /></Avatar>);
    expect(component.length).toBe(1);
    // TODO -> Assert something specific here
  });

  it('Should render proper status and size', () => {
    let component;
    let newProps;

    const props = {
      type: 'circle',
      size: 'xsmall',
      withStatus: true,
      status: 'live',
      src: 'http://i46.tinypic.com/sexbb8.png',
      alt: 'An avatar image',
    };

    component = mountWithTheme(<Avatar {...props}></Avatar>);

    expect(component.length).toBe(1);
    expect(component.find('StyledAvatarStatus').props().size).toBe('xsmall');
    expect(component.find('StyledAvatarStatus').props().status).toBe('live');

    newProps = { ...props, status: 'offline', size: 'small' };
    component = mountWithTheme(<Avatar {...newProps}></Avatar>);

    expect(component.length).toBe(1);
    expect(component.find('StyledAvatarStatus').props().size).toBe('small');
    expect(component.find('StyledAvatarStatus').props().status).toBe('offline');

    newProps = { ...props, status: 'online', size: 'medium' };
    component = mountWithTheme(<Avatar {...newProps}></Avatar>);

    expect(component.length).toBe(1);
    expect(component.find('StyledAvatarStatus').props().size).toBe('medium');
    expect(component.find('StyledAvatarStatus').props().status).toBe('online');

    newProps = {
      ...props, status: 'sleeping', size: 'large', type: 'rounded',
    };
    component = mountWithTheme(<Avatar {...newProps}></Avatar>);

    expect(component.length).toBe(1);
    expect(component.find('StyledAvatarStatus').props().size).toBe('large');
    expect(component.find('StyledAvatarStatus').props().status).toBe('sleeping');
  });
});
