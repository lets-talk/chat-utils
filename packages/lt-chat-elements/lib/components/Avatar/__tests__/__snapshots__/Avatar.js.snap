// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render with data and children case (Component children) 1`] = `
<StyledAvatarContainer
  className="LT-Avatar-Container"
  size="medium"
>
  <StyledChildrenContainer
    color="#FEFEFE"
    size="medium"
    type="circle"
  >
    <SimpleComponent />
    <StyledAvatarStatus
      size="medium"
      status="online"
    >
       
    </StyledAvatarStatus>
  </StyledChildrenContainer>
</StyledAvatarContainer>
`;

exports[`Avatar component should render with data and children case (String children) 1`] = `
<StyledAvatarContainer
  className="LT-Avatar-Container"
  size="xsmall"
>
  <StyledChildrenContainer
    color="#FEFEFE"
    size="xsmall"
    type="circle"
  >
    S
    <StyledAvatarStatus
      size="xsmall"
      status="live"
    >
       
    </StyledAvatarStatus>
  </StyledChildrenContainer>
</StyledAvatarContainer>
`;

exports[`Avatar component should render with data and circle case 1`] = `
<StyledAvatarContainer
  className="LT-Avatar-Container"
  size="small"
>
  <StyledImageContainer>
    <styled.img
      alt="An avatar image"
      size="small"
      sizes={null}
      src="http://i46.tinypic.com/sexbb8.png"
      srcSet={null}
      type="circle"
    />
    <ImageInnerDiv />
    <StyledAvatarStatus
      size="small"
      status="online"
    >
       
    </StyledAvatarStatus>
  </StyledImageContainer>
</StyledAvatarContainer>
`;

exports[`Avatar component should render without data 1`] = `
<StyledAvatarContainer
  className="LT-Avatar-Container"
  size="medium"
/>
`;
